Device[] devices
Application[] applications
string machine
time timestamp
